server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailGet.scala

```diff
@@ -78,12 +78,13 @@ case class EmailGetRequest(accountId: AccountId,
         attachments = bodyStructure.attachments,
         headers = asEmailHeaders(mime4JMessage.getHeader),
         bodyValues = bodyValues,
-        messageId = extractMessageId(mime4JMessage))
+        messageId = extractMessageId(mime4JMessage, "Message-Id"),
+        inReplyTo = extractMessageId(mime4JMessage, "In-Reply-To"))
     }
   }
 
-  private def extractMessageId(mime4JMessage: Message): Option[List[HeaderMessageId]] =
-    Option(mime4JMessage.getHeader.getFields("Message-Id"))
+  private def extractMessageId(mime4JMessage: Message, fieldName: String): Option[List[HeaderMessageId]] =
```

Comment:
should something like `extractHeaderField`?

Reply:
No, we extract header fields that hold a MessageId value => my name is suitable.